diff-containers/src/Data/Map/Diff/Strict/Internal.hs

```diff
@@ -199,10 +227,51 @@ null (Diff m) = Map.null m
 size :: Diff k v -> Int
 size (Diff m) = Map.size m
 
+-- | A positive diff contains only @'Insert'@s and @'Delete'@s. A negative diff
+-- can also contain @'UnsafeAntiInsert'@s and @'UnsafeAntiDelete'@s, which makes
+-- applying diffs unsafe.
+isPositive :: Diff k v -> Bool
+isPositive (Diff m) = all (isPositiveDiffHistory . toDiffHistory) m
+
+-- | Check if a diff history is positive, which means it can only contain
+-- @'Insert'@s and @'Delete'@s.
+isPositiveDiffHistory :: DiffHistory v -> Bool
```

Comment:
Consider adding `isPositiveDiffHistory` and `isNormalDiffHistory` as local definitions if we only use them in one function.

Reply:
The tests use these definitions, and I think it would be useful to keep exposing it to the user even though it is an Internal module